buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterFormationTasks.groovy

```diff
@@ -347,9 +347,19 @@ class ClusterFormationTasks {
 
         Task writeConfig = project.tasks.create(name: name, type: DefaultTask, dependsOn: setup)
         writeConfig.doFirst {
-            String unicastTransportUri = node.config.unicastTransportUri(seedNode, node, project.ant)
-            if (unicastTransportUri != null) {
-                esConfig['discovery.zen.ping.unicast.hosts'] = "\"${unicastTransportUri}\""
+            if (node.nodeVersion.onOrAfter("6.5.0-SNAPSHOT")) {
+                // Don't force discovery provider if one is set by the test cluster specs already
+                if (esConfig.containsKey('discovery.zen.hosts_provider') == false) {
+                    esConfig['discovery.zen.hosts_provider'] = 'file'
+                    esConfig['discovery.zen.ping.unicast.hosts'] = []
```

Comment:
I'm concerned that this should also be set to `[]` even if `discovery.zen.hosts_provider` is set externally. The places where I can see it being set externally are `azure`, `ec2` and `gce` and in all these cases we don't want to be probing local ports `9300-9304` too.

Reply:
Makes sense will do :)